server/bitbucketclient/factory.go

```diff
@@ -0,0 +1,24 @@
+package bitbucketclient
+
+import (
+	"fmt"
+	"net/http"
+
+	bitbucketv1 "github.com/gfleury/go-bitbucket-v1"
+)
+
+type ClientConfiguration struct {
+	SelfHostedURL    string
+	SelfHostedAPIURL string
+	APIClient        *bitbucketv1.APIClient
+	OAuthClient      *http.Client
+
+	LogError func(msg string, keyValuePairs ...interface{})
+}
+
+func GetBitbucketClient(clientType string, config ClientConfiguration) (Client, error) {
+	if clientType == "server" {
+		return newServerClient(config), nil
+	}
+	return nil, fmt.Errorf("wrong client passed")
+}
```

Comment:
Just an idea, but I'm thinking having two different functions for creating the two clients may be more Go-idiomatic. So having a `NewBBServerClient()` function for instance to create the BB Server client